src/Impostor.Server/Events/Game/BeforeGameCreatedEvent.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Impostor.Api.Events;
+using Impostor.Api.Innersloth;
+using Impostor.Server.Net.Redirector;
+using Impostor.Server.Net.State;
+
+namespace Impostor.Server.Events
+{
+    public class BeforeGameCreatedEvent : IBeforeGameCreatedEvent
+    {
+        private readonly ICollection<int> _gameKeys;
+
+        public BeforeGameCreatedEvent(ICollection<int> gameKeys)
+        {
+            _gameKeys = gameKeys;
+        }
+
+        public string? GameCode { get; private set; }
+
+        public bool TryToSetCode(string gameCode)
+        {
+            if (gameCode == null)
```

Comment:
already implemented in [GameCodeParser](https://github.com/Impostor/Impostor/blob/dev/src/Impostor.Api/Innersloth/GameCodeParser.cs)?

Reply:
> I didn't see anything like that in `GameCodeParser` :thinking:

Reply:
Explained better in new comments